<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from nj130_sensor.urdf.xacro        | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="right_nj130" xmlns:xacro="http://ros.org/wiki/xacro">
  <!-- Place the stand in the world -->
  <link name="world"/>
  <!-- Stand macros -->
  <link name="stand">
    <visual>
      <geometry>
        <mesh filename="package://itri_support/meshes/nj130/stand_base.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://itri_support/meshes/nj130/stand_base.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
  </link>
  <!-- place world reference -->
  <joint name="Base_to_stand" type="fixed">
    <parent link="world"/>
    <child link="stand"/>
    <origin rpy="0 0 0" xyz="0 0 0.3"/>
  </joint>
  <!-- meshes: main serial chain -->
  <link name="base_link">
    <inertial>
      <mass value="774.8899"/>
      <orgin rpy="0 0 0" xyz="0.0000 -0.0362 0.1323"/>
      <inertia ixx="33.5119" ixy="-0.0049" ixz="-0.0003" iyy="25.6574" iyz="-0.0198" izz="47.9022"/>
    </inertial>
    <collision name="base_link">
      <origin rpy="0  0  -1.57079632679" xyz="0  0  0"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/nj130/base.STL" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <visual name="base_link">
      <origin rpy="0  0  -1.57079632679" xyz="0  0  0"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/nj130/base.STL" scale="1.0 1.0 1.0"/>
      </geometry>
    </visual>
  </link>
  <link name="link1">
    <inertial>
      <mass value="413.782044"/>
      <origin rpy="0 0 0" xyz="-0.012546 0.242656 0.296132"/>
      <inertia ixx="32.763251" ixy="0.429344" ixz="0.165571" iyy="32.475622" iyz="10.371882" izz="28.635599"/>
    </inertial>
    <collision name="link1">
      <origin rpy="0  0  0" xyz="0  0  0"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/nj130/link1.STL" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <visual name="link1">
      <origin rpy="0  0  0" xyz="0  0  0"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/nj130/link1.STL" scale="1.0 1.0 1.0"/>
      </geometry>
    </visual>
  </link>
  <link name="link2">
    <inertial>
      <mass value="179.412103"/>
      <origin rpy="0 0 0" xyz="0.023928 0.523998 0.024200"/>
      <inertia ixx="24.944316" ixy="0.056517" ixz="0.054721" iyy="2.680444" iyz="0.401816" izz="24.097291"/>
    </inertial>
    <collision name="link2">
      <origin rpy="0  0  0" xyz="0  0  0"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/nj130/link2.STL" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <visual name="link2">
      <origin rpy="0  0  0" xyz="0  0  0"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/nj130/link2.STL" scale="1.0 1.0 1.0"/>
      </geometry>
    </visual>
  </link>
  <link name="link3">
    <inertial>
      <mass value="198.716462"/>
      <origin rpy="0 0 0" xyz="0.216194 -0.018066 -0.014441"/>
      <inertia ixx="6.146618" ixy="-0.569827" ixz="-0.167201" iyy="4.306393" iyz="0.883908" izz="4.125998"/>
    </inertial>
    <collision name="link3">
      <origin rpy="0  0  0" xyz="0  0  0"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/nj130/link3.STL" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <visual name="link3">
      <origin rpy="0  0  0" xyz="0  0  0"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/nj130/link3.STL" scale="1.0 1.0 1.0"/>
      </geometry>
    </visual>
  </link>
  <link name="link4">
    <inertial>
      <mass value="11.691993"/>
      <orgin rpy="0 0 0" xyz="-0.000010 -0.000140 0.413256"/>
      <inertia ixx="0.394124" ixy="0.000010" ixz="-0.000032" iyy="0.393863" iyz="0.000464" izz="0.046591"/>
    </inertial>
    <collision name="link4">
      <origin rpy="0  0  0" xyz="0  0  0"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/nj130/link4.STL" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <visual name="link4">
      <origin rpy="0  0  0" xyz="0  0  0"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/nj130/link4.STL" scale="1.0 1.0 1.0"/>
      </geometry>
    </visual>
  </link>
  <link name="link5">
    <inertial>
      <mass value="17.204093"/>
      <orgin rpy="0 0 0" xyz="-0.000967 -0.098289 -0.000467"/>
      <inertia ixx="0.105372" ixy="-0.001203" ixz="0.000079" iyy="0.067485" iyz="0.002341" izz="0.089779"/>
    </inertial>
    <collision name="link5">
      <origin rpy="0  0  0" xyz="0  0  0"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/nj130/link5.STL" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <visual name="link5">
      <origin rpy="0  0  0" xyz="0  0  0"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/nj130/link5.STL" scale="1.0 1.0 1.0"/>
      </geometry>
    </visual>
  </link>
  <link name="link6">
    <inertial>
      <mass value="3.766890"/>
      <orgin rpy="0 0 0" xyz="0.000000 -0.000095 -0.017652"/>
      <inertia ixx="0.006572" ixy="0.0" ixz="0.0" iyy="0.007020" iyz="0.000035" izz="0.012705"/>
    </inertial>
    <collision name="link6">
      <origin rpy="0  0  0" xyz="0  0  0"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/nj130/link6.STL" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <visual name="link6">
      <origin rpy="0  0  0" xyz="0  0  0"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/nj130/link6.STL" scale="1.0 1.0 1.0"/>
      </geometry>
    </visual>
  </link>
  <!-- end of link list -->
  <!-- joints: main serial chain -->
  <joint name="joint1" type="revolute">
    <parent link="base_link"/>
    <child link="link1"/>
    <origin rpy="0  0  -1.57079632679" xyz="0  0  0.331"/>
    <axis xyz="0  0  -1"/>
    <dynamics damping="0.7" friction="0.7"/>
    <limit effort="30.0" lower="-3.14159265359" upper="3.14159265359" velocity="1.919862"/>
  </joint>
  <joint name="joint2" type="revolute">
    <parent link="link1"/>
    <child link="link2"/>
    <origin rpy="1.57079632679  0  1.57079632679" xyz="0  0.35  0.499"/>
    <axis xyz="0  0  -1"/>
    <dynamics damping="0.7" friction="0.7"/>
    <limit effort="30.0" lower="-1.308997" upper="1.658" velocity="1.919862"/>
  </joint>
  <joint name="joint3" type="revolute">
    <parent link="link2"/>
    <child link="link3"/>
    <origin rpy="0  0  0" xyz="0  1.16  0"/>
    <axis xyz="0  0  -1"/>
    <dynamics damping="0.7" friction="0.7"/>
    <limit effort="30.0" lower="-4.468" upper="-0.017453" velocity="1.919862"/>
  </joint>
  <joint name="joint4" type="revolute">
    <parent link="link3"/>
    <child link="link4"/>
    <origin rpy="1.57079632679  0  0" xyz="0.25  0  0"/>
    <axis xyz="0  0  -1"/>
    <dynamics damping="0.7" friction="0.7"/>
    <limit effort="30.0" lower="-4.8869" upper="4.8869" velocity="3.31612"/>
  </joint>
  <joint name="joint5" type="revolute">
    <parent link="link4"/>
    <child link="link5"/>
    <origin rpy="-1.57079632679  0  0" xyz="0  0  1.10558"/>
    <axis xyz="0  0  -1"/>
    <dynamics damping="0.7" friction="0.7"/>
    <limit effort="30.0" lower="-2.09439510239" upper="2.09439510239" velocity="3.31612"/>
  </joint>
  <joint name="joint6" type="revolute">
    <parent link="link5"/>
    <child link="link6"/>
    <origin rpy="1.57079632679  3.14159265359  0" xyz="0  -0.21  0"/>
    <axis xyz="0  0  -1"/>
    <dynamics damping="0.7" friction="0.7"/>
    <!-- limit effort="30.0" lower="${-15*pi}" upper="${15*pi}" velocity="4.01425"/ -->
    <limit effort="30.0" lower="-3.14159265359" upper="3.14159265359" velocity="4.01425"/>
  </joint>
  <!-- end of joint list -->
  <!-- ROS-Industrial 'flange' frame: attachment point for EEF models -->
  <link name="flange"/>
  <joint name="link6_flange" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="link6"/>
    <child link="flange"/>
  </joint>
  <joint name="stand_to_arm" type="fixed">
    <parent link="stand"/>
    <child link="base_link"/>
    <origin rpy="0 0 0" xyz="0 0 0.3"/>
    <!-- limit effort="30.0" lower="-1.0" upper="1.0" velocity="0.3"/ -->
  </joint>
  <link name="hand">
    <inertial>
      <mass value="0.73"/>
      <origin rpy="0 0 0" xyz="0 0.0015244 -0.9275912"/>
      <inertia ixx="0.00278560230025" ixy="0.0" ixz="0.0" iyy="0.000400033405336" iyz="0.0" izz="0.00256378041832"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.09"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/gripper/visual/hand.stl"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.09"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/gripper/collision/hand.stl"/>
      </geometry>
    </collision>
  </link>
  <!-- link name="leftfinger">
      <inertial>
        <mass value="0.1" />
        <origin xyz="0 0.0145644 0.0227941" rpy="0 0 0" />
        <inertia ixx="3.01220925051e-05" ixy="0.0" ixz="0.0" iyy="2.95873808038e-05" iyz="0.0" izz="6.95125211657e-06" />
      </inertial>
      <visual>
        <origin xyz="${xyz}" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://itri_support/meshes/gripper/visual/finger.dae"/>
        </geometry>
      </visual>
      <collision>
        <origin xyz="${xyz}" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://itri_support/meshes/gripper/collision/finger.stl"/>
        </geometry>
      </collision>
    </link>
    <link name="rightfinger">
      <inertial>
        <mass value="0.1" />
        <origin xyz="0 -0.0145644 0.0227941" rpy="0 0 ${pi}" />
        <inertia ixx="3.01220925051e-05" ixy="0.0" ixz="0.0" iyy="2.95873808038e-05" iyz="0.0" izz="6.95125211657e-06" />
      </inertial>
      <visual>
        <origin xyz="${xyz}" rpy="0 0 ${pi}"/>
        <geometry>
          <mesh filename="package://itri_support/meshes/gripper/visual/finger.dae"/>
        </geometry>
      </visual>
      <collision>
        <origin xyz="${xyz}" rpy="0 0 ${pi}"/>
        <geometry>
          <mesh filename="package://itri_support/meshes/gripper/collision/finger.stl"/>
        </geometry>
      </collision>
    </link -->
  <link name="rightfinger">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.01 0 0"/>
      <mass value="0.1"/>
      <inertia ixx="0.002" ixy="0" ixz="0" iyy="0.002" iyz="0" izz="0.002"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0.11685 0"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/gripper/visual/r_gripper_finger_link.stl"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0.11685 0"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/gripper/collision/r_gripper_finger_link.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="leftfinger">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.01 0 0"/>
      <mass value="0.1"/>
      <inertia ixx="0.002" ixy="0" ixz="0" iyy="0.002" iyz="0" izz="0.002"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 -0.11685 0"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/gripper/visual/l_gripper_finger_link.stl"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 -0.11685 0"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/gripper/collision/l_gripper_finger_link.stl"/>
      </geometry>
    </collision>
  </link>
  <!-- joint name="${prefix}finger_joint1" type="prismatic">
      <parent link="${prefix}hand"/>
      <child link="${prefix}leftfinger"/>
      <origin xyz="0 0 0.0584" rpy="0 0 0"/>
      <axis xyz="0 1 0"/>
      <limit effort="20" lower="0.0" upper="0.05" velocity="0.05"/>
    </joint>
    <joint name="${prefix}finger_joint2" type="prismatic">
      <parent link="${prefix}hand"/>
      <child link="${prefix}rightfinger"/>
      <origin xyz="0 0 0.0584" rpy="0 0 0"/>
      <axis xyz="0 -1 0"/>
      <limit effort="20" lower="0.0" upper="0.05" velocity="0.05"/>
    </joint -->
  <joint name="finger_joint1" type="prismatic">
    <origin rpy="0 -1.57079632679 0" xyz="0 0 0"/>
    <parent link="hand"/>
    <child link="rightfinger"/>
    <axis xyz="0 1 0"/>
    <limit effort="60" lower="0.0" upper="0.05" velocity="0.05"/>
  </joint>
  <joint name="finger_joint2" type="prismatic">
    <origin rpy="0 -1.57079632679 0" xyz="0 0 0"/>
    <parent link="hand"/>
    <child link="leftfinger"/>
    <axis xyz="0 -1 0"/>
    <limit effort="60" lower="0.0" upper="0.05" velocity="0.05"/>
  </joint>
  <joint name="hand_joint" type="fixed">
    <parent link="ft_sensor_link"/>
    <child link="hand"/>
    <origin rpy="0 0 0" xyz="0 0 0.095"/>
  </joint>
  <!-- gazebo reference="${prefix}hand_joint">
        <disableFixedJointLumping>true</disableFixedJointLumping>
      </gazebo -->
  <gazebo reference="hand">
    <material>Gazebo/Green</material>
    <self_collide>true</self_collide>
  </gazebo>
  <gazebo reference="leftfinger">
    <material>Gazebo/Turquoise</material>
    <self_collide>true</self_collide>
  </gazebo>
  <gazebo reference="rightfinger">
    <material>Gazebo/Turquoise</material>
    <self_collide>true</self_collide>
  </gazebo>
  <transmission name="finger_joint1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="finger_joint1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="finger_joint1_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="finger_joint2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="finger_joint2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="finger_joint2_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="ft_sensor_link">
    <inertial>
      <mass value="0.01"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/sensor/ft_sensor.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://itri_support/meshes/sensor/ft_sensor.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </visual>
  </link>
  <joint name="ft_sensor_joint" type="fixed">
    <parent link="link6"/>
    <child link="ft_sensor_link"/>
    <origin rpy="0 0 0" xyz="0 0 0.01"/>
  </joint>
  <gazebo reference="ft_sensor_link">
    <material>Gazebo/Black</material>
    <gravity>true</gravity>
    <sensor name="imu_sensor" type="imu">
      <always_on>true</always_on>
      <update_rate>1000</update_rate>
      <visualize>true</visualize>
      <topic>__default_topic__</topic>
      <plugin filename="libgazebo_ros_imu_sensor.so" name="imu_plugin">
        <topicName>imu</topicName>
        <bodyName>ft_sensor_link</bodyName>
        <updateRateHZ>200.0</updateRateHZ>
        <gaussianNoise>0.0</gaussianNoise>
        <xyzOffset>0 0 0</xyzOffset>
        <rpyOffset>0 0 0</rpyOffset>
        <frameName>ft_sensor_link</frameName>
      </plugin>
      <pose>0 0 0 0 0 0</pose>
    </sensor>
  </gazebo>
  <gazebo reference="ft_sensor_joint">
    <disableFixedJointLumping>true</disableFixedJointLumping>
    <provideFeedback>true</provideFeedback>
  </gazebo>
  <gazebo>
    <plugin filename="libgazebo_ros_ft_sensor.so" name="ft_sensor">
      <updateRate>1000.0</updateRate>
      <topicName>ft_raw</topicName>
      <jointName>ft_sensor_joint</jointName>
      <gaussianNoise>0.0</gaussianNoise>
    </plugin>
  </gazebo>
  <gazebo reference="base_link">
    <material>Gazebo/RedBright</material>
  </gazebo>
  <gazebo reference="link1">
    <material>Gazebo/RedBright</material>
  </gazebo>
  <gazebo reference="link2">
    <material>Gazebo/RedBright</material>
  </gazebo>
  <gazebo reference="link3">
    <material>Gazebo/RedBright</material>
  </gazebo>
  <gazebo reference="link4">
    <material>Gazebo/RedBright</material>
  </gazebo>
  <gazebo reference="link5">
    <material>Gazebo/RedBright</material>
  </gazebo>
  <gazebo reference="link6">
    <material>Gazebo/RedBright</material>
  </gazebo>
  <transmission name="trans1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="trans2">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="trans3">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint3">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor3">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="trans4">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint4">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor4">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="trans5">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint5">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor5">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="trans6">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint6">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor6">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <!-- gazebo ros control -->
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="gazebo_ros_control">
      <robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType>
      <legacyModeNS>true</legacyModeNS>
    </plugin>
  </gazebo>
  <!-- gazebo grasp fix plugin -->
  <gazebo>
    <plugin filename="libgazebo_grasp_fix.so" name="gazebo_grasp_fix">
      <arm>
        <arm_name>nj130</arm_name>
        <palm_link>link6</palm_link>
        <gripper_link>leftfinger</gripper_link>
        <gripper_link>rightfinger</gripper_link>
      </arm>
      <forces_angle_tolerance>90</forces_angle_tolerance>
      <update_rate>4</update_rate>
      <grip_count_threshold>3</grip_count_threshold>
      <max_grip_count>8</max_grip_count>
      <release_tolerance>0.001</release_tolerance>
      <disable_collisions_on_attach>false</disable_collisions_on_attach>
      <contact_topic>__default_topic__</contact_topic>
    </plugin>
  </gazebo>
</robot>
